web_ui/src/components/UsageBillingPage.tsx

```diff
@@ -70,7 +106,688 @@ function UsageAndBillingContainer({ children }: { children: React.ReactNode }) {
+      <Modal.Header closeButton>
+        <Modal.Title>Start Trial</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <Form
+          onSubmit={(e: React.FormEvent) => {
+            e.preventDefault()
+            startTrial()
+          }}>
+          <Form.Group controlId="formBasicEmail">
+            <Form.Label>Notification Email</Form.Label>
+            <Form.Control
+              type="email"
+              placeholder="Enter email"
+              required
+              value={email}
+              onChange={(x: React.ChangeEvent<HTMLInputElement>) =>
+                setEmail(x.target.value)
+              }
+            />
+            <Form.Text className="text-muted">
+              We’ll send you trial reminders at this email address.
+            </Form.Text>
+          </Form.Group>
+          <Button variant="primary" type="submit" disabled={loading}>
+            {loading ? "Loading" : "Begin Trial"}
+          </Button>
+          {error && <Form.Text className="text-danger">{error}</Form.Text>}
+          <Form.Text className="text-muted">
+            Your trial will expire 14 days after start.
+          </Form.Text>
+        </Form>
+      </Modal.Body>
+    </Modal>
+  )
+}
+
+interface IStartSubscriptionModalProps {
+  readonly show: boolean
+  readonly onClose: () => void
+  readonly seatUsage: number
+}
+function StartSubscriptionModal({
+  show,
+  onClose,
+  seatUsage,
+}: IStartSubscriptionModalProps) {
+  const [seats, setSeats] = React.useState(1)
+  const [loading, setLoading] = React.useState(false)
+  const [error, setError] = React.useState("")
+
+  function setSubscription() {
+    setLoading(true)
+    setError("")
+
+    teamApi(Current.api.startCheckout, {
+      seatCount: seats,
+    }).then(async res => {
+      if (res.ok) {
+        const stripe = await loadStripe(res.data.stripePublishableApiKey)
+        if (stripe == null) {
+          setError("Failed to load Stripe")
+          return
+        }
+        const { error } = await stripe.redirectToCheckout({
+          sessionId: res.data.stripeCheckoutSessionId,
+        })
+        setError(error.message || "error redirecting to checkout")
+      } else {
+        setError("Failed to start checkout")
+      }
+      setLoading(false)
+    })
+  }
+
+  const costCents = seats * settings.monthlyCost
+  return (
+    <Modal show={show} onHide={onClose}>
+      <Modal.Header closeButton>
+        <Modal.Title>Start Subscription</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <Form
+          onSubmit={(e: React.FormEvent) => {
+            e.preventDefault()
+            setSubscription()
+          }}>
+          <Form.Group controlId="formBasicEmail">
+            <Form.Label>Seats</Form.Label>
+            <Form.Control
+              type="number"
+              required
+              min="1"
+              placeholder="Enter seat count"
+              value={String(seats)}
```

Comment:
if we keep the value as a string, then we can remove this
```suggestion
              value={seats}
```